From 0f216ce274b0a30271dad53244fa1085c527efde Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20=C5=A0tetiar?= <ynezz@true.cz>
Date: Tue, 7 Mar 2023 17:22:20 +0100
Subject: [PATCH] mod-dhcp-uci: dnsmasq_cmd: fix missing execvp null pointer
 array termination
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Experincing quite strange DHCPv4 issues and tracked them down to
following:

 prplOS procd: instance 1�1�9�vi�� not found

 tr181-dhcpv4client -D  | dnsmasq restart 1�1�9�vi��
 /bin/sh /sbin/service dnsmasq restart 1�1�9�vi��

 dhcpv4-manager -D  | dnsmasq restart 1�1�9�vi��
 /bin/sh /sbin/service dnsmasq restart 1�1�9�vi��

It seems like amxp_subproc_vstart_wait() uses execvp() under the hood,
which expects passed command argv array of pointers to be terminated by
a null pointer, so lets add that missing null pointer sentinel.

Signed-off-by: Petr Štetiar <ynezz@true.cz>
---
 mod-dhcp-uci/src/mod_dhcp_uci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mod-dhcp-uci/src/mod_dhcp_uci.c b/mod-dhcp-uci/src/mod_dhcp_uci.c
index 47a20f161d4a..a64c31dd17e2 100644
--- a/mod-dhcp-uci/src/mod_dhcp_uci.c
+++ b/mod-dhcp-uci/src/mod_dhcp_uci.c
@@ -102,7 +102,7 @@ static struct _dhcp_options opts[] = {
  * @return 0 on success
  */
 static int dnsmasq_cmd(const char* cmd) {
-    const char* args[] = {NULL, NULL, NULL};
+    const char* args[] = {NULL, NULL, NULL, NULL};
     int rc = 1;
     amxp_subproc_t* subproc = NULL;
 
